File: 3872-find-most-frequent-vowel-and-consonant/find-most-frequent-vowel-and-consonant.js
/**
 * @param {string} s
 * @return {number}
 */
var maxFreqSum = function (s) {
    const n = s.length;
    const freqArr = new Array(26).fill(0);

    for (let i = 0; i < n; i++) {
        const charIndex = s.charCodeAt(i) - 'a'.charCodeAt(0);
        freqArr[charIndex]++;
    }

    const vowels = new Set(['a', 'e', 'i', 'o', 'u']);
    let vowelMax = -Infinity;
    let consonantMax = -Infinity;

    for (let i = 0; i < 26; i++) {
        const char = String.fromCharCode(i + 'a'.charCodeAt(0));
        if (vowels.has(char)) {
            vowelMax = Math.max(vowelMax, freqArr[i]);
        } else {
            consonantMax = Math.max(consonantMax, freqArr[i]);
        }
    }

    return vowelMax + consonantMax;

};

File: 3872-find-most-frequent-vowel-and-consonant/find-most-frequent-vowel-and-consonant.java
class Solution {
    public int maxFreqSum(String s) {
        int n = s.length();
        int[] freqArr = new int[26];

        // Count frequency of each character
        for (int i = 0; i < n; i++) {
            int charIndex = s.charAt(i) - 'a';
            freqArr[charIndex]++;
        }

        // Define vowels
        String vowels = "aeiou";
        int vowelMax = Integer.MIN_VALUE;
        int consonantMax = Integer.MIN_VALUE;

        // Find max frequency among vowels and consonants
        for (int i = 0; i < 26; i++) {
            char ch = (char) (i + 'a');
            if (vowels.indexOf(ch) != -1) { // it's a vowel
                vowelMax = Math.max(vowelMax, freqArr[i]);
            } else { // it's a consonant
                consonantMax = Math.max(consonantMax, freqArr[i]);
            }
        }

        return vowelMax + consonantMax;
    }
}
